src/parser/asts/functions.c

```
@@ -174,15 +180,37 @@ AST_NODE* parseASMFunctionDeclaration(struct LexerResult result, int index) {
+		if(t.type != NUMBER) {
 			return NULL;
 		}
 
 		buff[buffIndex] = strtol(t.value, NULL, 16);
+		buffIndex++;
 	}
 
 	node->endingIndex = index;
-	node->value = (char*) buff;
+
+	buff = realloc(buff, sizeof(uint8_t) * buffIndex);
```

cppcheck (2.10-2)

[error] Common realloc mistake

(memleakOnRealloc)